### PR TITLE
Update devconsole empty state pages

### PR DESCRIPTION
**Pull request by rottencandy**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5857

**Analysis / Root cause**: 
Some pages in devconsole were not consistent for when the user does not have access to any project.

**Solution Description**: 
- Show "getting started" notification when user with no project access navigates to topology. Do not show Add page.
- Show projects list in search page when user is not in a specific namespace.
- Show getting started notification on Add page for user with no project access.

**Screen shots / Gifs for design review**: 
Topology - no accessible projects:

Search - project not selected:

+Add - no accessible projects:

@openshift/team-devconsole-ux 

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

/kind bug

### CODE REVIEW

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Got TS errors around the `withStartGuide`

Also, your Search page is missing out on the project selector/secondary masthead:
image

**Comment by rottencandy** [comment]
> Got TS errors around the `withStartGuide`
> 
> Also, your Search page is missing out on the project selector/secondary masthead

Fixed Type errors, and added namespace selector:

**Comment by rohitkrai03** [comment]
@rottencandy Seems like this UX doc suggests we should reuse current topology empty state with the getting started section on top. But that will have an issue because we have options like topology quick add and add to project links in topology empty states enabled. In this case when there are no projects, those options cannot be enabled. @bgliwa01 WDYT?

File: frontend/packages/topology/src/components/page/TopologyPage.tsx [hunk 1]
```diff
@@ -105,27 +100,25 @@ const TopologyPage: React.FC<TopologyPageProps> = ({
             viewType === TopologyViewType.graph ? 'topology-graph-page' : 'topology-list-page'
           }
         >
-          <ProjectsExistWrapper title={t('topology~Topology')} projects={projects}>
```
[the remark this answers is not in the document]
Comment (rottencandy) [reply frontend/packages/topology/src/components/page/TopologyPage.tsx:108]: Updated Add page to use `withStartGuide` and deleted `ProjectsExistWrapper`.

**Review by andrewballantyne — APPROVED** [review]

**Review by rohitkrai03 — APPROVED** [review]
/lgtm

**Comment by bgliwa01** [comment]
The secondary masthead with the project selector should be above the blue box. Thanks!

File: frontend/packages/dev-console/src/components/add/AddPage.tsx [hunk 2]
```diff
@@ -25,28 +25,24 @@ const AddPage: React.FC<AddPageProps> = ({ match }) => {
         <title data-test-id="page-title">{`+${t('devconsole~Add')}`}</title>
       </Helmet>
       <NamespacedPage>
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/add/AddPage.tsx:27]: We need this component to be not inside the `withStartGuide` HOC.

File: frontend/packages/topology/src/components/page/TopologyPage.tsx [hunk 3]
```diff
@@ -34,7 +32,7 @@ interface TopologyPageProps {
   defaultViewType?: TopologyViewType;
 }
 
-const TopologyPage: React.FC<TopologyPageProps> = ({
+export const TopologyPage: React.FC<TopologyPageProps> = ({
```
Comment (andrewballantyne) [inline frontend/packages/topology/src/components/page/TopologyPage.tsx:35]: Same problem for this guy, but I can't put a comment on line 90 🙂

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/packages/topology/src/components/page/TopologyPage.tsx (hunk 3 above)
Comment (rottencandy) [reply frontend/packages/topology/src/components/page/TopologyPage.tsx:65]: Updated. Moved blue box to below project dropdown in both places.

File: frontend/packages/dev-console/src/components/SearchPage.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+              .
+            </Trans>
+          )}
+        </CreateProjectListPage>
+      )}
+    </NamespacedPage>
+  );
+};
+
+export default Search;
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/SearchPage.tsx:40]: Did we not want the `withStartGuide` here?

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Only question I have is do we want to setup Search like the other pages?

File: frontend/packages/dev-console/src/components/SearchPage.tsx (hunk 4 above)
Comment (rottencandy) [reply frontend/packages/dev-console/src/components/SearchPage.tsx:45]: Updated.

**Review by andrewballantyne — APPROVED** [review]
/lgtm

**Pull request merged**